pkg/social/generic_oauth.go

```diff
@@ -193,6 +193,11 @@ func (s *SocialGenericOAuth) UserInfo(client *http.Client, token *oauth2.Token)
 		if err != nil {
 			return nil, fmt.Errorf("Error decoding user info JSON: %s", err)
 		}
+
+		err = json.Unmarshal(response.Body, &data.Attributes)
```

Comment:
this looks like a big breaking change

Reply:
before we looks at at attributes property now we deserialize everything on the root level into this map

Reply:
I agree this is a breaking change but :

- this attributes map seems not to be in the Oauth UserInfo schema  : https://connect2id.com/products/server/docs/api/userinfo
- I found nothing that could deserialize it in grafana code and decided to use a simple map of strings.

Reply:
I think many oauth solution have custom attributes in this field , but I was not involved in adding it, you will have to trace that back with git history of when this was added and why.

Reply:
Yeah, this may. We have custom OAuth stuff over here.

Reply:
Thank you for your feedback. 
If we don't want any breaking change, I can create a second field "JsonAttributes" to get the full json properties, and adapt the extractEmail function. Would this be acceptable for you ?

Reply:
There are better ways to do that, you can have a raw property or something to deserialize an object and additional fields, can you google it? I am on mobile right now

Reply:
There are few solutions presented here : 
https://stackoverflow.com/questions/33436730/unmarshal-json-with-unknown-fields

We could unmarshal somewhere and then refill manually the existing Attributes fields, but this could give some breaking changes too.

Reply:
Hello,
After few days off and training, I was able to propose a new non breaking solution.
Can you have a look at it ?

Reply:
Any update / remark ?